### PR TITLE
don't add missing replicated node from kill packet

### PR DESCRIPTION
**Pull request by birarda**
- fixed a bug where upstream replicated agents would be added from kill packets

### CODE REVIEW

**Comment by birarda** [comment]
### Test Plan

This is an extreme edge case that will only be consistently testable by modifying the source.

Change an upstream avatar mixer to replicate only kill packets for its broadcasted users. Confirm that in the downstream avatar mixer you never see an upstream agent added.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1031/HighFidelity-Beta-PR10791-24a4898a59f8b8842f4ac7187cbb2e80904fb783.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1031/HighFidelity-Beta-PR10791-24a4898a59f8b8842f4ac7187cbb2e80904fb783.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1506/HighFidelity-Beta-PR10791-98ccfa80a229fa37aa2f7c21dd0f1079583f24e5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1506/HighFidelity-Beta-PR10791-98ccfa80a229fa37aa2f7c21dd0f1079583f24e5.exe

**Comment by birarda** [comment]
@Atlante45 and I tested this locally using the following diff:

```
diff --git a/assignment-client/src/avatars/AvatarMixerSlave.cpp b/assignment-client/src/avatars/AvatarMixerSlave.cpp
index 7e37f58..6630cc0 100644
--- a/assignment-client/src/avatars/AvatarMixerSlave.cpp
+++ b/assignment-client/src/avatars/AvatarMixerSlave.cpp
@@ -440,6 +440,8 @@ uint64_t REBROADCAST_IDENTITY_TO_DOWNSTREAM_EVERY_US = 5 * 1000 * 1000;
 void AvatarMixerSlave::broadcastAvatarDataToDownstreamMixer(const SharedNodePointer& node) {
     _stats.downstreamMixersBroadcastedTo++;
 
+    return;
+
```

I ran that patch on my avatar mixer and @Atlante45 ran a downstream avatar mixer from both master and this PR. From this PR, the above patch did not cause his avatar mixer to add the killed node (before killing it immediately). On master, the above patch causes his avatar mixer to add the killed node and then remove it.

QA pass.
